editor/animation_track_editor.cpp

```diff
@@ -5753,7 +5755,35 @@ void AnimationTrackEditor::_update_key_edit() {
 		key_edit->hint = _find_hint_for_track(key_edit->track, np);
 		key_edit->base = np;
 
-		EditorNode::get_singleton()->push_item(key_edit);
+		// Highlighting skeleton3D bone gizmo if keyframe is modifying a bone
+		Animation::TrackType track_type = animation->track_get_type(key_edit->track);
+		Skeleton3DEditor *se = Skeleton3DEditor::get_singleton();
+
+		// Only show selected gizmos if we have a skeleton3DEditor instance and it is in editor mode
+		if (track_type == Animation::TYPE_ROTATION_3D && se && se->is_edit_mode()) {
```

Comment:
Shouldn't TYPE_POSITION_3D and TYPE_SCALE_3D also be handled?

Reply:
It can be. The feature request just mentioned TYPE_ROTATION_3D, but I can definitely make sure it handles all of these cases.